src/docfx/Errors.cs

```diff
@@ -159,9 +159,10 @@ public static Error GitCloneFailed(string url, IEnumerable<string> branches)
 
         /// <summary>
         /// Yaml header defined in article.md isn't an object.
+        /// The line should always be 2 since the file should always start with "---"
         /// </summary>
-        public static Error YamlHeaderNotObject(bool isArray)
-            => new Error(ErrorLevel.Warning, "yaml-header-not-object", $"Expect yaml header to be an object, but got {(isArray ? "an array" : "a scalar")}");
+        public static Error YamlHeaderNotObject(bool isArray, string file)
+            => new Error(ErrorLevel.Warning, "yaml-header-not-object", $"Expect yaml header to be an object, but got {(isArray ? "an array" : "a scalar")}", new SourceInfo(file, 2, 1));
```

Comment:
haha

Reply:
YAML header will always be the second line :)
https://github.com/lunet-io/markdig/blob/master/src/Markdig.Tests/Specs/YamlSpecs.md